samza-core/src/main/java/org/apache/samza/clustermanager/SamzaApplicationState.java

```diff
@@ -110,7 +110,17 @@
    */
   public final ConcurrentMap<String, SamzaResource> runningProcessors = new ConcurrentHashMap<>(0);
 
-   /**
+  /**
+   *  Map of the failed Samza processor ID to resource status of the last attempted of the container.
+   *  This map is only used when {@link org.apache.samza.config.ClusterManagerConfig#CLUSTER_MANAGER_CONTAINER_FAIL_JOB_AFTER_RETRIES}
+   *  is set to false, this map tracks the containers which have exhausted all retires for restart and JobCoordinator is
+   *  no longer attempting to restart this container
+   *
+   *  Modified by both the AMRMCallbackThread and the ContainerAllocator thread
+   */
+  public final ConcurrentHashMap<String, SamzaResourceStatus> failedProcessors = new ConcurrentHashMap<>(0);
```

Comment:
This class already has a map called 
```
/**
   * ContainerStatuses of failed containers.	   * ContainerStatuses of failed containers.
   */	 
  public final ConcurrentMap<String, SamzaResourceStatus> failedContainersStatus = new ConcurrentHashMap<>();	 
```

Could we please reuse that?

Also the variable type can just be a Map<,> doesnt need to be ConcurrentHashMap

Reply:
As long as we use one of the two. 
Its also used by YarnClusterResourceManager for updating "completed" containers, 
which are containers that exit after finishing gracefully.

Reply:
thats YarnAppState not SamzaApplicationState